manifests/init.pp

```diff
@@ -32,11 +50,15 @@
     create_resources('systemd::service_limits', $service_limits)
   }
 
-  if $manage_resolved {
+  if $manage_resolved and $facts['systemd_internal_services']['systemd-resolved.service'] {
```

Comment:
This is going to fail hard if the fact could not be populated for some reason.

Should probably be: `if $mange_resolved and $facts['systemd_internal_services'] and $facts['systemd_internal_services']['systemd-resolved.service']